File: 이상욱/BOJ/m4w9/BOJ4358.java
package m4w9;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;

public class BOJ4358 {
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		
		HashMap<String, Integer> map = new HashMap<>();
		
		int tot = 0;
		
		while(true) {
			
			String tree = br.readLine();
			if(tree == null) {
				break;
			}
			
			tot++;
			
			map.put(tree, map.getOrDefault(tree, 0) + 1);
			
		}
		
		StringBuilder sb = new StringBuilder();
		
		ArrayList<String> list = new ArrayList<>();
		
		for(String key : map.keySet()) {
			list.add(key);
		}
		
		Collections.sort(list);
		
		for(String key : list) {
			String tName = key;
			int cnt = map.get(key);
			double p = cnt * 100.0 / tot;
			
			sb.append(tName+" "+String.format("%.4f", p)+"\n");
		}
		
		bw.write(sb.toString());
		bw.flush();
		br.close();
		bw.close();
		
	}
}

File: 이상욱/BOJ/m4w9/BOJ4358.kt
package m4w9

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.lang.StringBuilder

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val map = hashMapOf<String,Int>()


    var tot = 0
    while (true){
        val tree = br.readLine()?:break

        tot++
        map[tree] = map.getOrDefault(tree,0) + 1
    }

    val sb = StringBuilder()

    val sortedMap = map.toSortedMap()

    for( t in sortedMap.keys ){
        val tName = t
        val cnt = map[t]

        if (cnt != null) {
            sb.append("$tName ${String.format("%.4f",cnt*100 / tot.toDouble())}\n")
        }
    }
    bw.write(sb.toString())
    bw.flush()
    br.close()
    bw.close()

}